etcd/watchmanager/types.go

```diff
@@ -35,7 +35,7 @@ type WatchManager interface {
 }
 
 // UpdateFn is called when an event on the watch channel happens
-type UpdateFn func(key string) error
+type UpdateFn func(key string, events []*clientv3.Event) error
```

Comment:
Or you could handle the withEvents and noEvents path separately in the updateFn and only share the mergeCache/updateWatch part?